kazuma.py

```diff
@@ -205,6 +237,64 @@ def processimage(tempsticker):
     im.save(tempsticker, "PNG")
     im.close()
 
+def process_vid(tempsticker):
+    """Change file to webm format with proper dimensions (atleast one side 512px)"""
+
+    ff = FFmpeg()
+    vid = cv2.VideoCapture(tempsticker)
+
+    if tempsticker.endswith('webm'):
+        webm_tempsticker = tempsticker
+    else:
+        webm_tempsticker = os.path.splitext(tempsticker)[0] + '.webm'
+
+    h = vid.get(cv2.CAP_PROP_FRAME_HEIGHT)
+    w = vid.get(cv2.CAP_PROP_FRAME_WIDTH)
+
+    if h >= w:
+        h = 512
+        w = -1
+    else:
+        w = 512
+        h = -1
+
+    op = ff.options(f"-i {tempsticker} -filter:v scale={w}:{h} -c:a copy -an {webm_tempsticker}")
+
+def check_vid(replymsg, tempsticker):
+    """Check if the file fulfill the requirements specified in https://core.telegram.org/stickers#video-stickers"""
+
+    fp = FFprobe(tempsticker)
+    
+    # Convert duration to seconds. 00:01:02.120 -> 62.12s
+    dur, milliseconds = fp.duration.split('.')
+    x = time.strptime(dur,'%H:%M:%S')
+    duration = datetime.timedelta(hours=x.tm_hour, minutes=x.tm_min, seconds=x.tm_sec, milliseconds=int(milliseconds)).total_seconds()
+
+    frame_rate = float(fp.fps)
+    size = os.path.getsize(tempsticker)
+
+    # Duration max 3s
+    if duration > 3:
+        replymsg.edit_text(s.REPLY_VID_DURATION_ERROR.format(duration))
+
+    # Frame rate max 30fps
+    elif frame_rate > 30:
```

Comment:
Should I make the user to send files less than 30fps or should I convert the frame rate to 30fps using ffmpeg if it's greater than 30fps.
I reckon latter would be much useful for users.

Reply:
Chose to do the latter